iosxr_setup.py

```diff
@@ -50,10 +68,22 @@ def setup_networking(self, host_ip=None, gateway=None, allow_root_login=False):
 
         # Determine if the image is a crypto/k9 image or not
         # This will be used to determine whether to configure ssh or not
-        xr1.send("run rpm -qa | grep k9sec")
-        time.sleep(2)
-        output = xr1.wait("[\$#]")
+        output = self.send_operns("rpm -qa | grep k9sec")
         k9 = re.search(r'-k9sec', output)
+        if k9:
+            xr1.log("Crypto k9 image detected")
+        else:
+            xr1.log("Non crypto k9 image detected")
+
+        # Determine if the image is a full or mini by searching for the mgbl rpm
+        # which only exists in the full image
+        # This will be used to configure features only available in a full image
+        output = self.send_operns("rpm -qa | grep mgbl")
+        full = re.search(r'-mgbl', output)
+        if full:
+            xr1.log("Full image detected")
```

Comment:
*Technically* this is checking whether mgbl rpm is installed (which is all we actually need for grpc) not whether it's a full image, so a mini + mgbl image would also meet this requirement - might be good to change "full" to "mgbl" throughout, but non-blocking.